clang/include/clang/Sema/Overload.h

```diff
@@ -876,6 +876,11 @@ class Sema;
     /// function pointer or reference (C++ [over.call.object]).
     FunctionDecl *Function;
 
+    /// LambdaName - When the OverloadCandidate is for a
+    /// lambda's operator(), points to the declaration of
+    /// the lambda variable.
+    VarDecl *LambdaName{nullptr};
```

Comment:
What do we do when this isn't present?  There are plenty of cases where we can have a lambda overload candidate without a name.  We probably need to make sure that there are tests for this feature that cover lambdas in template-packs, as well as immediately invoked ones not stored to a variable.

Reply:
@zyn0217 unfortunately removing the default initializer breaks a bunch of tests--need to check to confirm but i believe it's due to the check that `LambdaDecl` is `nullptr` here: https://github.com/llvm/llvm-project/blob/d8140f9013f47094a159fd2918128d9252ca3280/clang/lib/Sema/SemaCodeComplete.cpp#L6130

i think the fix would just be setting `LambdaDecl` to `nullptr` everywhere we make an `OverloadCandidate` for a non-lambda, but the `nullptr` default initializer seemed cleaner... let me know if not using a default initializer is preferred.

Reply:
The pack lambda I was thinking about was something like:

```
template <typename ... Lambdas>
void bar(Lambdas... Ls) {
  Ls()...;
}

void caller() {
bar([]{return 1;}, []{return 1.1;}, []{return "1";}); // where each are mildly different/etc

```

It seems it would be weird to me to print 'Ls' for each, since as they are distinct, particularly if that causes a diagnostic.

Reply:
thanks for the example!
i don't think this patch affects behavior in this scenario. the current behavior (preserved in this patch) is not to produce any signature help. i believe this is because within `bar()` it is not known that `Ls` has a call operator. unless i'm misunderstanding?
<img width="438" alt="Screenshot 2024-09-09 at 20 05 32" src="https://github.com/user-attachments/assets/f19431e0-3764-4530-8cf7-22cec4f8f228">

Reply:
Yep, exactly.  In the instantiation of 'bar', the various calls to the 'Ls'.  Also, I was more concerned at one point because i thought this affected diagnostics, but I don't see code that does that anymore.